### PR TITLE
Use a framework target instead of a static library

### PR DESCRIPTION
**Pull request by eliperkins**
This removes the static library target from the project and adds in a framework target, introducing compatibility with `use_frameworks!` from Cocoapods and Carthage support.

This also introduces a few quality of life improvements like:
- Correcting header imports/removing circular imports
- Bumping a few of our testing dependencies
- Bumping our bundled Ruby dependencies
- Removing our `VENFail()` macro in favor of Expecta's `failure()`
- Bumping to iOS 8.0 for our deployment target
- Removing CMDQueryStringComponents from our unit tests
- A new, more advanced, state of the art `.gitignore`

### CODE REVIEW

File: VENCoreUnitTests/Models/Users/VENUserSpec.m [hunk 1]
```diff
@@ -71,21 +71,21 @@
 
 
 describe(@"Initialization", ^{
-
+
```
Comment (dasmer) [inline VENCoreUnitTests/Models/Users/VENUserSpec.m]: it looks like a bunch of empty space lines were added?

File: VENCoreUnitTests/Utilities/EXPMatchers+Venmo.h [hunk 2]
```diff
@@ -1,3 +1,5 @@
 #import <Expecta/Expecta.h>
 
-EXPMatcherInterface(match, (id pattern))
\ No newline at end of file
+EXPMatcherInterface(regexMatch, (id pattern))
+
+#define regexMatch regexMatch
```
Comment (dasmer) [inline VENCoreUnitTests/Utilities/EXPMatchers+Venmo.h]: newline EOF

**Comment by dasmer** [comment]
Thanks for doing this. This really looks great. A few nits and :green_apple:

File: VENCoreUnitTests/Models/Users/VENUserSpec.m (hunk 1 above)
Comment (eliperkins) [inline VENCoreUnitTests/Models/Users/VENUserSpec.m]: Ugh. I just `ctrl-I`'ed in Xcode. I guess that's what I get... I'll see if I can clean this up.

File: VENCoreUnitTests/Utilities/EXPMatchers+Venmo.h (hunk 2 above)
Comment (eliperkins) [inline VENCoreUnitTests/Utilities/EXPMatchers+Venmo.h]: Got it in  3ed5bdb

File: VENCoreUnitTests/Models/Users/VENUserSpec.m (hunk 1 above)
Comment (eliperkins) [inline VENCoreUnitTests/Models/Users/VENUserSpec.m]: Cleaned it up!

**Comment by dasmer** [comment]
:green_apple:

**Pull request merged**